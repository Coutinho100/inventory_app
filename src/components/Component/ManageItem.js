import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity, ImageBackground } from 'react-native';


const ManageItem = props => {
    let sn = props.serialNo
    // let sn = props.serialNo.replace('i', '');
    let styler = sn % 2 === 0 ? styles.mealItem : styles.mealItem2;

    return (
        <View style={styler}>
            <TouchableOpacity onPress={props.onSelectItem} >

                <View style={{ ...styles.mealRow, ...styles.mealDetails }}>
                    <Text> {sn}</Text>
                    <Text numberOfLines={1}>{props.title}</Text>
                    <Text> {props.quantity}pcs</Text>
                    <Text> {props.company.toUpperCase()}</Text>
                    <Text> Rs. {props.price}/PC</Text>
                </View>
            </TouchableOpacity>
        </View >
    )
}

const styles = StyleSheet.create({
    titleContainer: {
        backgroundColor: 'rgba(0,0,0,0.5)',
        paddingVertical: 10,
        paddingHorizontal: 12,
    }
    , title: {
        fontFamily: 'open-sans-bold',
        fontSize: 20,
        // color: 'white',
        textAlign: 'center'
    },
    mealItem: {
        width: '100%',
        height: 50,
        backgroundColor: '#e5e5e5',
        borderRadius: 10,
        overflow: 'hidden',
        justifyContent: 'center'

    }, mealItem2: {
        width: '100%',
        height: 50,
        backgroundColor: '#ffffff',
        borderRadius: 10,
        overflow: 'hidden',
        justifyContent: 'center'
    },
    mealRow: {
        flexDirection: 'row',
        alignItems: 'center'
    }
    ,

    mealDetails: {
        // height: '15%',
        paddingHorizontal: 10,
        justifyContent: 'space-between',
        alignItems: 'center'
    },

})

export default ManageItem;