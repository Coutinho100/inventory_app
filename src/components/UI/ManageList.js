import React from 'react';
import { View, FlatList, StyleSheet, Text } from 'react-native';
import ManageItem from '../Component/ManageItem';

const ManageList = props => {

    const renderInventoryItem = (itemData, index) => {
        // console.log("manage list ", itemData)

        return (
            <ManageItem
                key={itemData.item._id}
                title={itemData.item.title}
                quantity={itemData.item.quantity}
                company={itemData.item.company}
                price={itemData.item.price}
                serialNo={itemData.index}
                onSelectItem={() => {
                    props.navigation.navigate({
                        routeName: 'AddItem',
                        params: {
                            itemId: itemData.item._id,
                            itemTitle: itemData.item.title,
                        }
                    }
                    );
                }}
            />
        );
    };
    return (
        <View style={styles.list}>
            <View style={styles.title}>
                <Text style={{...styles.topic, flex:0.2}}> SN</Text>
                <Text style={{...styles.topic, flex:0.4}}>ITEM NAME</Text>
                <Text style={{...styles.topic, flex:0.4}}> QUANTITY</Text>
                <Text style={{...styles.topic, flex:0.4}}> COMPANY</Text>
                <Text style={{...styles.topic, flex:0.3}}> PRICE</Text>
            </View>
            <FlatList
                data={props.listData}
                keyExtractor={(item, index) => item._id}
                renderItem={renderInventoryItem}
                style={{ width: '100%' }}
            />
        </View>
    );
};

const styles = StyleSheet.create({
    list: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        padding: 15
    },
    title: {
        flexDirection: 'row',
        width: '100%',
        justifyContent: 'space-between',
        paddingHorizontal: 8,
        paddingBottom: 10,
    },
    topic: {
        fontFamily: 'open-sans-bold',
        fontSize: 16
    }
});
export default ManageList;