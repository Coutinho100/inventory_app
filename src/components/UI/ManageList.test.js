import React from 'react';
import ManageList from './ManageList';
import {
    configure,
    shallow
} from 'enzyme';
import { Text, View, FlatList } from 'react-native';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import test_function, {findByTestAttr} from '../../_test_/componenets/test_function'
import ManageItem from '../Component/ManageItem';
// JestHook.mock('expo-font');

configure({
    adapter: new EnzymeAdapter
})
const function_click=jest.fn()



const dummy_data = {
listData:["a","b","c"],
item:{
    image:"image.png",
    quantity:4,
    title:"hello",
    price:34,
    company:"abc",
    _id:"id2"
},
catTitle:"titles",
navigation:{navigate:function_click}
}

let extra=
describe('<ManageList />', () => {
  it("ManageItem in ManageList", () =>
        {
            const wrapper = shallow(<ManageList  {...dummy_data} />)

            let flatList= wrapper.find(FlatList)
            expect(wrapper.find(FlatList).length).toEqual(1);            
            expect(flatList.prop("data")).toEqual(["a","b","c"])
            let dive = flatList.props().renderItem(dummy_data)
            wrapper.update()
            expect(dive.type).toEqual(ManageItem)
            expect(dive.props.title).toEqual("hello")
            dive.props.onSelectItem()
            expect(function_click).toHaveBeenCalledTimes(1);
            expect(function_click).toHaveBeenCalledWith({
                "routeName": "AddItem",               
                 params: {
                itemId: "id2",
                itemTitle: "hello",

            }
            })          
            // let inventoryItem= findByTestAttr(wrapper,"inventoryItem")
            // expect(inventoryItem).toHaveLength(1)

        
            });
        });

test_function(["<ManageList/>",5,2,1],['renders correctly','renders the length of view'],[<ManageList />,Text,View,FlatList],extra)

