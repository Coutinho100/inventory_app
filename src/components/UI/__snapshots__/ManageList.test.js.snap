// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManageList/> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 10,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "open-sans-bold",
          "fontSize": 16,
        }
      }
    >
       SN
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "open-sans-bold",
          "fontSize": 16,
        }
      }
    >
      ITEM NAME
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "open-sans-bold",
          "fontSize": 16,
        }
      }
    >
       QUANTITY
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "open-sans-bold",
          "fontSize": 16,
        }
      }
    >
       COMPANY
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "open-sans-bold",
          "fontSize": 16,
        }
      }
    >
       PRICE
    </Text>
  </View>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
